packages/framework/src/Model/Product/ProductElasticsearchProvider.php

```diff
@@ -76,4 +76,34 @@ public function getSellableProductArrayByUuids(array $productUuids): array
             $this->filterQueryFactory->createSellableProductsByProductUuidsFilter($productUuids),
         );
     }
+
+    /**
+     * @param int[][] $productsIds
+     * @return array
+     */
+    public function getBatchedVisibleByProductIds(array $productsIds): array
```

Comment:
Batching is used only in frontend API, so I think it should not be present in framework. This applies for all batching methods introduced to framework